src/rust/multi.bato/src/lib.rs

```diff
@@ -0,0 +1,108 @@
+#![no_std]
+#![feature(pattern)]
+
+use aidoku::{
+	error::Result, prelude::*, std::defaults::defaults_get, std::net::HttpMethod,
+	std::net::Request, std::String, std::Vec, Chapter, DeepLink, Filter, Listing, Manga,
+	MangaPageResult, Page,
+};
+
+mod crypto;
+mod evpkdf;
+mod parser;
+mod substring;
+
+#[get_manga_list]
+fn get_manga_list(filters: Vec<Filter>, page: i32) -> Result<MangaPageResult> {
+	let mut result: Vec<Manga> = Vec::new();
+
+	let (url, search) = parser::get_filtered_url(filters, page);
+	let html = Request::new(url.as_str(), HttpMethod::Get).html();
+	if search {
+		parser::parse_search(html, &mut result);
+	} else {
+		parser::parse_listing(html, &mut result);
+	}
+
+	if !result.is_empty() {
+		Ok(MangaPageResult {
+			manga: result,
+			has_more: true,
+		})
+	} else {
+		Ok(MangaPageResult {
+			manga: result,
+			has_more: false,
+		})
+	}
+}
+
+#[get_manga_listing]
+fn get_manga_listing(listing: Listing, page: i32) -> Result<MangaPageResult> {
+	let mut url = String::new();
+	if let Ok(url_str) = defaults_get("sourceURL").as_string() {
+		url.push_str(url_str.read().as_str());
+	}
+	let mut result: Vec<Manga> = Vec::new();
+	if listing.name == "Popular" {
+		parser::get_list_url(&mut url, "views_a.za", page);
+	}
+	if listing.name == "Latest" {
+		parser::get_list_url(&mut url, "update.za", page);
+	}
+	if listing.name == "New Titles" {
+		parser::get_list_url(&mut url, "create.za", page);
+	}
+	let html = Request::new(url.as_str(), HttpMethod::Get).html();
+	parser::parse_listing(html, &mut result);
```

Comment:
why doesn't this just return the result?

Reply:
Fixed in [b2d09b9](https://github.com/Skittyblock/aidoku-community-sources/pull/114/commits/b2d09b99f745f01a41774e062663c26dc7255565)

Reply:
I wrote the get_manga_list before this function and don't feel like changing it.